CosineDistance.lua

```diff
@@ -8,6 +8,9 @@ end
 function CosineDistance:updateOutput(input)
    local input1, input2 = input[1], input[2]
 
+   if not input1:isContiguous() then input1 = input1:clone() end
```

Comment:
Thanks @fmassa . Updated, now put a local function wrap there to reuse the buffer for the backward pass as suggested. Tested on my end. See if this is good.